### PR TITLE
refactor: removed attribute/protocol mapper logic

### PR DESCRIPTION
**Pull request by ziccardi**
## Description
Removed the attribute/protocol mapper logic

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has created for changes required on the client side

### CODE REVIEW

File: pkg/services/keycloak.go [hunk 1]
```diff
@@ -602,10 +585,6 @@ func (kc *keycloakService) registerAgentServiceAccount(clusterId string, service
 	}
 	if hasRole == nil {
 		glog.V(10).Infof("Client %s has no role %s, adding", serviceAccountId, roleName)
-		addRoleErr := kc.kcClient.AddRealmRoleToUser(accessToken, *serviceAccountUser.ID, *role)
-		if addRoleErr != nil {
-			return nil, errors.NewWithCause(errors.ErrorGeneral, addRoleErr, "failed to add role to agent service account")
-		}
 	}
```
Comment (machi1990) [inline pkg/services/keycloak.go:588]: The whole condition can go away I guess

File: pkg/services/keycloak.go [hunk 2]
```diff
@@ -553,29 +553,12 @@ func (kc *keycloakService) registerAgentServiceAccount(clusterId string, service
...
-		return nil, err
-	}
-
-	protocolMapper := kc.kcClient.CreateProtocolMapperConfig(clusterId)
-
-	attributes := map[string]string{
-		clusterId: agentClusterId,
-	}
-
-	if kc.GetConfig().KeycloakClientExpire {
```
Comment (machi1990) [inline pkg/services/keycloak.go:567]: let's make sure the `KeycloakClientExpire` config is removed if it is not used anywhere else

**Review by machi1990 — COMMENTED** [review]
The change lgtm.

I'll let @akoserwal  have a look on it. Plus a member of the connectors team /cc @lburgazzoli @dhirajsb

File: pkg/services/keycloak.go (hunk 2 above)
Comment (dhirajsb) [reply pkg/services/keycloak.go:567]: Are we getting rid of auto expiring clients that were used for integration tests?

**Review by akoserwal — APPROVED** [review]

File: pkg/services/keycloak.go (hunk 2 above)
Comment (machi1990) [reply pkg/services/keycloak.go:567]: > Are we getting rid of auto expiring clients that were used for integration tests?

/cc @akoserwal @ziccardi

Comment (dhirajsb) [reply pkg/services/keycloak.go:567]: Thanks @akoserwal

**Pull request merged**